projects/kit/components/tabs/tabs.directive.ts

```diff
@@ -65,4 +87,33 @@ export class TuiTabsDirective implements AfterViewChecked {
             nativeElement.setAttribute(`tabIndex`, active ? `0` : `-1`);
         });
     }
+
+    ngOnDestroy(): void {
+        this.windowRef.clearInterval(this.scrollingId);
+    }
+
+    private safelyScrollToActiveElement(): void {
+        this.ngZone.runOutsideAngular(() => {
```

Comment:
If we use `ngOnChanges` instead of a setter, do we still need timeout? I feel like the has got to be a way to do it without all this mess.

Reply:
` do we still need timeout? ` yes, because without timeout our `tab.offsetLeft` return `0`

Reply:
And ngAfterViewChecked is called every time